ci_framework/roles/ci_setup/tasks/repos.yml

```diff
@@ -0,0 +1,62 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+
+- name: Ensure repos are enabled in the system using rhsm.
+  become: true
+  tags:
+    - bootstrap
+    - packages
+  when: ansible_distribution == 'RedHat'
+  block:
+    - name: Check rhsm status
+      ansible.builtin.command:
+        cmd: "subscription-manager status"
+
+    - name: Gather the repos to be enabled
+      ansible.builtin.set_fact:
+        _repos: >-
+          {{
+            cifmw_ci_setup_default_repos +
+            (cifmw_ci_setup_extra_repos | default([]))
+          }}
```

Comment:
Below are the code execution of cifmw_ci_setup_extra_repos and cifmw_ci_setup_extra_repos
```
ci_framework/roles/ci_setup/vars/redhat.yml
cifmw_ci_setup_default_repos:
  - 'rhel-*-baseos-rpms'
  - 'rhel-*-appstream-rpms'
  
  ci_framework/roles/devscripts/tasks/sub_tasks/11_packages.yml
    cifmw_ci_setup_extra_repos:
      - 'codeready-builder-*-x86_64-rpms'  
      
ci_framework/roles/ci_setup/tasks/repos.yml
    - name: Gather the repos to be enabled
      ansible.builtin.set_fact:
        _repos: >-
          {{
            cifmw_ci_setup_default_repos +
            (cifmw_ci_setup_extra_repos | default([]))
          }}
    - name: Enabling the required repositories.
      community.general.rhsm_repository:
        name: "{{ item }}"
        state: enabled
      loop: "{{ _repos }}"
ci_framework/roles/ci_setup/tasks/cleanup.yml
- name: Disable the repos enabled in rhsm
  become: true
  when:
    - cifmw_ci_setup_extra_repos is defined
    - cifmw_ci_setup_extra_repos | length > 0
    - ansible_distribution == 'RedHat'
  community.general.rhsm_repository:
    name: "{{ item }}"
    state: disabled
  loop: "{{ cifmw_ci_setup_extra_repos }}"
  ```
  As we can see `cifmw_ci_setup_default_repos` and `cifmw_ci_setup_extra_repos` vars for setting rhsm rhel repos.
  Can we merge remove `cifmw_ci_setup_extra_repos` and merge the content of `cifmw_ci_setup_extra_repos` into `cifmw_ci_setup_default_repos` and also rename `cifmw_ci_setup_default_repos` to `cifmw_ci_setup_rhel_rhsm_repos`?
  in Devscript role, we are enabling code-ready containers, I think we can move them under ci-setup role.
  Let me know your thoughts.

Reply:
Thank you. Yes, it does make sense to rename `cifwm_ci_setup_rhel_rhsm_repos`. Let me do that change.

With regards enabling `crb` by default, I am bit hesitant... I believe `ci_reproducer` requires it... in that case, let me enable it. We can modify later.

Let me push another commit.